buildNpmPackage: document dontNpmBuild option (#230157)
NixOS/nixpkgs: doc/languages-frameworks/javascript.section.md

* `npmDepsHash`: The output hash of the dependencies for this project. Can be calculated in advance with [`prefetch-npm-deps`](#javascript-buildNpmPackage-prefetch-npm-deps).
* `makeCacheWritable`: Whether to make the cache writable prior to installing dependencies. Don't set this unless npm tries to write to the cache directory, as it can slow down the build.
* `npmBuildScript`: The script to run to build the project. Defaults to `"build"`.

> * `dontNpmBuild`: Option to disable running the build script, set to `true` if it's missing in the project. Defaults to `false`.

Reviewer: I'd argue we should also mention you can disable this by setting `buildPhase` explicitly. cc @lilyinstarlight for her thoughts

Reviewer 2: That's not a bad idea. I'm not sure how best to phrase or whether it should even go here or somewhere else in the docs, though

We are also missing docs for `dontNpmInstall`

> * `dontNpmBuild`: Option to disable running the build script, set to `true` if it's missing in the project. Defaults to `false`.

Reviewer 2: ```suggestion
* `dontNpmBuild`: Option to disable running the build script. Set to `true` if the package does not have a build script. Defaults to `false`.
```

Maybe adjust wording a little to make it clearer? This PR otherwise looks good

* `npmDepsHash`: The output hash of the dependencies for this project. Can be calculated in advance with [`prefetch-npm-deps`](#javascript-buildNpmPackage-prefetch-npm-deps).
* `makeCacheWritable`: Whether to make the cache writable prior to installing dependencies. Don't set this unless npm tries to write to the cache directory, as it can slow down the build.
* `npmBuildScript`: The script to run to build the project. Defaults to `"build"`.
* `dontNpmBuild`: Option to disable running the build script. Set to `true` if the package does not have a build script. Defaults to `false`. Alternatively, setting `buildPhase` explicitly also disables this.

> * `dontNpmInstall`: Option to disable running `npm install`. Defaults to `false`.

Author: Thanks for the feedback, fixed.
